### PR TITLE
TAJO-1497: RPC client does not share a connection

### PR DESCRIPTION
**Pull request by jinossy**
I've change RpcConnectionPool to RpcConnectionManager and remove the release codes.
Because netty rpc does not need pooling.

### CODE REVIEW

**Comment by jihoonson** [comment]
I think this patch is very important to make Tajo more stable. So, I'll review it first.

**Comment by jinossy** [comment]
I've rename to RpcClientManager

File: tajo-client/src/main/java/org/apache/tajo/client/SessionConnection.java [hunk 1]
```diff
@@ -286,13 +286,13 @@ public void close() {
...
     try {
-      client = connPool.getConnection(getTajoMasterAddr(), TajoMasterClientProtocol.class, false);
+      client = manager.getConnection(getTajoMasterAddr(), TajoMasterClientProtocol.class, false);
       TajoMasterClientProtocolService.BlockingInterface tajoMaster = client.getStub();
       tajoMaster.removeSession(null, sessionId);
     } catch (Throwable e) {
       // ignore
     } finally {
-      connPool.releaseConnection(client);
+      manager.cleanup(client);
```
Comment (jihoonson) [inline tajo-client/src/main/java/org/apache/tajo/client/SessionConnection.java]: This is a static function.

Comment (jinossy) [inline tajo-client/src/main/java/org/apache/tajo/client/SessionConnection.java]: Thank you for nice finding

File: tajo-rpc/tajo-rpc-protobuf/src/main/java/org/apache/tajo/rpc/ServerCallable.java [hunk 2]
```diff
@@ -33,17 +33,17 @@
...
 
   public abstract T call(NettyClientBase client) throws Exception;
 
-  public ServerCallable(RpcConnectionPool connPool,  InetSocketAddress addr, Class<?> protocol, boolean asyncMode) {
+  public ServerCallable(RpcClientManager connPool,  InetSocketAddress addr, Class<?> protocol, boolean asyncMode) {
     this(connPool, addr, protocol, asyncMode, false);
   }
 
-  public ServerCallable(RpcConnectionPool connPool, InetSocketAddress addr, Class<?> protocol,
+  public ServerCallable(RpcClientManager manager, InetSocketAddress addr, Class<?> protocol,
```
Comment (jihoonson) [inline tajo-rpc/tajo-rpc-protobuf/src/main/java/org/apache/tajo/rpc/ServerCallable.java]: This constructor seems to be not used anymore. How about removing it?

Comment (jinossy) [inline tajo-rpc/tajo-rpc-protobuf/src/main/java/org/apache/tajo/rpc/ServerCallable.java]: I've remove constructor

**Comment by jinossy** [comment]
I've enable idle timeout. If all requests is done and client is idle state, client will be closed
```
public static final int RPC_IDLE_TIMEOUT = 43200; // 12 hour
```

**Comment by jihoonson** [comment]
+1
thanks for the nice patch!
I think that we can make our rpc cleaner with this patch.

**Pull request closed without merging**